appsmithorg/appsmith
# feat: gsheet form config changes

## Description
As a part of Limiting Google Sheets access project, we are modifying the UI of datasource configuration form for google sheets datasource as per [figma link](https://www.figma.com/file/TcFhqEbAc8ymHTRF5wR1qv/Limited-GSheet-Access?node-id=7%3A8&t=FN3j084OyCErlu67-0) This PR contains the code changes required for this new flow. Please check https://github.com/appsmithorg/appsmith/issues/20159 for more information. 
This improvement was suggested in form config for feature flag, I have created https://github.com/appsmithorg/appsmith/issues/20925 to track it, will be fixing it.

Steps to test:

1. Create new google sheet datasource, you should see first option in the dropdown as `Read/Write | Selected Sheets` selected by default.
2. In this case if we click on save and authorise, it should take us to google accounts page, where we can select the account.
3. Once account is selected, we should land on permissions page, where we can check the new permission of `See, edit, create, and delete only the specific Google Drive files you use with this app`, click on allow.
4. This will take us back to datasource config page in appsmith.
5. Other two options in dropdown are for providing Read/Write or Read only access to all files in google drive. For Read Only option we should see permissions as before, where as for read/write option we should see two permissions: `See, edit, create, and delete all of your Google Drive files` and `https://www.googleapis.com/auth/spreadsheets`.
7. Since this feature is hidden behind feature flag, we would also need to test the flow for normal user and ensure that current implementation is working as is without any issues


Fixes #20158 , #20159 
> if no issue exists, please create an issue and ask the maintainers about this first


## Type of change

- New feature (non-breaking change which adds functionality)


## How Has This Been Tested?

- Manual
- Cypress

### Test Plan
> Add Testsmith test cases links that relate to this PR

### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)


## Checklist:
### Dev activity


### QA activity:
